pkg/controller/process.go

```diff
@@ -114,5 +116,42 @@ func (c AppGwIngressController) Process(event events.Event) error {
 	glog.V(3).Info("cache: Updated with latest applied config.")
 	c.updateCache(&appGw)
 
+	// update ingresses with appgw gateway ip address
+	c.updateIngressStatus(generatedAppGw, cbCtx, event)
+
 	return nil
 }
+
+func (c AppGwIngressController) updateIngressStatus(appGw *n.ApplicationGateway, cbCtx *appgw.ConfigBuilderContext, event events.Event) {
+	addList := cbCtx.IngressList
+	removeList := make([]*v1beta1.Ingress, 0)
+	if ingress, ok := event.Value.(*v1beta1.Ingress); ok && event.Type == events.Update && !k8scontext.IsIngressApplicationGateway(ingress) {
+		removeList = append(removeList, ingress)
+	}
+
+	c.k8sContext.SetIngressStatuses(addList, removeList, c.getIPAddress(appGw, cbCtx))
+}
+
+func (c AppGwIngressController) getIPAddress(appGw *n.ApplicationGateway, cbCtx *appgw.ConfigBuilderContext) string {
+	ctx := context.Background()
+	for _, frontendIPConfiguration := range *appGw.FrontendIPConfigurations {
+		if cbCtx.EnvVariables.UsePrivateIP == "true" && frontendIPConfiguration.PrivateIPAddress != nil {
+			return *frontendIPConfiguration.PrivateIPAddress
+		} else if cbCtx.EnvVariables.UsePrivateIP != "true" && frontendIPConfiguration.PublicIPAddress != nil {
+			// parse publicIp resourceId
+			subscriptionID := GetSubscriptionIDFromResourceID(*frontendIPConfiguration.PublicIPAddress.ID)
+			resourceGroup := GetResourceGroupFromResourceID(*frontendIPConfiguration.PublicIPAddress.ID)
+			resourceName := GetResourceNameFromResourceID(*frontendIPConfiguration.PublicIPAddress.ID)
```

Comment:
I see that the goal is to extract `subscriptionID`, `resourceGroup`, and `resourceName` from the `frontendIPConfiguration.PublicIPAddress.ID` string.

I think the regexes are a smart way to parse the strings, but perhaps unnecessary since we know the format of the string.  Arguably it would be simpler to split on `/` and get the values from the corresponding positions, which are stable.


```go

func xyz(resourceID) (string, string, string) {
	split := strings.Split(resourceID, "/")
	return split[1], split[3], split[5]
}
subscriptionID, resourceGroup, ResourceName := xyz("/a/b/c/x/y/z")
```